tests/MongoDB.Driver.Tests/Linq/MongoQueryableTests.cs

```diff
@@ -1136,6 +1133,33 @@ public void Select_followed_by_group()
                 "{ $group: { _id: '$First', Stuff: { $push: { Id: '$Id', Second: '$Second' } } } }");
         }
 
+
+        [Fact]
+        public void Select_with_using_top_level_expression_input_parameters_inside_children_levels()
+        {
+            var query = CreateQuery()
+                .Select(g => new { Result = g.G.Where(g1 => g1.N > g.G.Min(g2 => g2.N))});
+
+            var commonResult = Assert(
+                query,
+                2,
+                "{ '$project' : { 'Result' : { '$filter' : { 'input' : '$G', 'as' : 'g1', 'cond' : { '$gt' : ['$$g1.N', { '$min' : '$G.N' }] } } }, '_id' : 0 } }");
+
+            commonResult[0].Result.Count().Should().Be(1);
+            commonResult[0].Result.First().N.Should().Be(2);
+            commonResult[1].Result.Count().Should().Be(1);
+            commonResult[1].Result.First().N.Should().Be(4);
+
+            // It looks like the below test generates a wrong query. The server returns the following error:
```

Comment:
As I wrote in the comment, it looks like the generated query is not correct. Should I create a new ticket for this error?

Reply:
that means that the `$map` is returning null, which is weird. The only way that should happen is if `G.B.S` doesn't exist in some document. If that's true, then we need to add in null checks, which really suck to have. But yes, probably file a new bug ticket.